Cannot find subscription in AzureUSGovernmentCloud

When trying to deploy the exporter to AzureUSGovernmentCloud, it's failing with status 404 - "The subscription '***REDACTED***' could not be found".
```
{"file":"main.go:62","func":"main","level":"info","msg":"starting azure-resourcemanager-exporter v21.10.1 (a5ae429; go1.17.2; by webdevops.io)"}
{"file":"main.go:63","func":"main","level":"info","msg":"{\"Logger\":{\"Debug\":true,\"Verbose\":true,\"LogJson\":true},\"Azure\":{\"Tenant\":\"***REDACTED***\",\"Environment\":\"AzureUSGovernmentCloud\",\"Subscription\":[\"***REDACTED***\"],\"Location\":[\"usgovtexas\"],\"ResourceGroupTags\":[\"owner\"],\"ResourceTags\":[\"owner\"]},\"Scrape\":{\"Time\":300000000000,\"TimeRateLimitRead\":0,\"TimeRateLimitWrite\":0,\"TimeExporter\":60000000000,\"TimeGeneral\":60000000000,\"TimeResource\":0,\"TimeQuota\":60000000000,\"TimeSecurity\":0,\"TimeResourceHealth\":0,\"TimeIam\":0,\"TimeGraph\":0,\"TimeCosts\":0},\"Graph\":{\"ApplicationFilter\":\"\"},\"Costs\":{\"Timeframe\":[\"MonthToDate\",\"YearToDate\"],\"Dimension\":[\"ResourceType\",\"ResourceLocation\"]},\"Portscan\":{\"Enabled\":false,\"Time\":10800000000000,\"Parallel\":2,\"Threads\":1000,\"Timeout\":5,\"PortRange\":[\"1-65535\"]},\"Metrics\":{\"ResourceIdLowercase\":false},\"Cache\":{\"Path\":\"\"},\"ServerBind\":\":8080\"}"}
{"file":"main.go:65","func":"main","level":"info","msg":"init Azure connection"}
{"file":"main.go:240","func":"initAzureConnection","level":"panic","msg":"subscriptions.Client#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code=\"SubscriptionNotFound\" Message=\"The subscription '***REDACTED***' could not be found.\""}
panic: (*logrus.Entry) 0xc0000d20e0

goroutine 1 [running]:
github.com/sirupsen/logrus.(*Entry).log(0xc00072b998, 0x0, {0xc00051c360, 0x102})
        /go/pkg/mod/github.com/sirupsen/logrus@v1.8.1/entry.go:259 +0x26f
github.com/sirupsen/logrus.(*Entry).Log(0xc0000d2070, 0x0, {0xc00072b998, 0x0, 0xc00018a120})
        /go/pkg/mod/github.com/sirupsen/logrus@v1.8.1/entry.go:293 +0x4f
github.com/sirupsen/logrus.(*Logger).Log(0xc0002bc000, 0x0, {0xc00072b998, 0x1, 0x1})
        /go/pkg/mod/github.com/sirupsen/logrus@v1.8.1/logger.go:198 +0x65
github.com/sirupsen/logrus.(*Logger).Panic(...)
        /go/pkg/mod/github.com/sirupsen/logrus@v1.8.1/logger.go:247
github.com/sirupsen/logrus.Panic(...)
        /go/pkg/mod/github.com/sirupsen/logrus@v1.8.1/exported.go:129
main.initAzureConnection()
        /go/src/github.com/webdevops/azure-resourcemanager-exporter/main.go:240 +0x535
main.main()
        /go/src/github.com/webdevops/azure-resourcemanager-exporter/main.go:66 +0x1b0
```

After having verified the subscription and tenant are correct, I am quite confident it's failing because of this piece of code:

https://github.com/webdevops/azure-resourcemanager-exporter/blob/09312c7eebfec371c3771aa72004f53a3e7c50f9/main.go#L220

Since BaseURI is not passed as argument, the Azure SDK uses the default one that matches Azure Public Cloud:

https://github.com/Azure/azure-sdk-for-go/blob/651a2232ee3a50a46adb298fbbfbd0efe7670db7/services/resources/mgmt/2021-01-01/subscriptions/subscriptions.go#L24-L27

--- a/main.go
+++ b/main.go
@@ -3,6 +3,15 @@ package main
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+
 	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/subscriptions"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
@@ -12,14 +21,6 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	log "github.com/sirupsen/logrus"
 	"github.com/webdevops/azure-resourcemanager-exporter/config"
-	"net/http"
-	"os"
-	"path"
-	"path/filepath"
-	"regexp"
-	"runtime"
-	"strconv"
-	"strings"
 )
 
 const (
@@ -217,7 +218,13 @@ func initAzureConnection() {
 	if err != nil {
 		log.Panic(err)
 	}
-	subscriptionsClient := subscriptions.NewClient()
+
+	azureEnvironment, err = azure.EnvironmentFromName(*opts.Azure.Environment)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	subscriptionsClient := subscriptions.NewClientWithBaseURI(azureEnvironment.ResourceManagerEndpoint)
 	subscriptionsClient.Authorizer = AzureAuthorizer
 
 	if len(opts.Azure.Subscription) == 0 {
@@ -241,11 +248,6 @@ func initAzureConnection() {
 			}
 			AzureSubscriptions = append(AzureSubscriptions, result)
 		}
-	}
-
-	azureEnvironment, err = azure.EnvironmentFromName(*opts.Azure.Environment)
-	if err != nil {
-		log.Panic(err)
 	}
 }
 

